### ISSUE TITLE
Generate Sobol points directly on the GPU

### ISSUE THREAD

**Issue by RagibZaman** [pytorch/pytorch#62648]
## 🚀 Feature

A method to generate a Tensor of Sobol points directly on the GPU, rather than the current implementation which creates the Sobol points on CPU which must then be transferred to GPU. 

## Motivation
Many PyTorch users utilizing Sobol points would to perform simulations with their GPU, so it would be natural to be able to generate the points on the GPU.

The 'curand' CUDA library already has a Sobol implementations that generates points directly on the GPU: https://docs.nvidia.com/cuda/curand/group__DEVICE.html.

**Comment by Balandat** [comment]
Yeah I agree that would be great to have. I don't know too much about the curand implementation, but one thing that might be hard to do is to get the same behavior on CPU and GPU (e.g. I would assume that different seeds generate different scrambled sequences, and that probably also the direction numbers or the deterministic sequence are different). But some discrepancy there is preferred to not having Sobol generation native on the GPU at all.
